src/Symbols/SymbolHelper.cpp

```diff
@@ -266,10 +264,41 @@ ErrorMessageOr<fs::path> SymbolHelper::FindSymbolsInCache(const fs::path& module
                                                           uint64_t expected_file_size) const {
   return FindSymbolsInCacheImpl(
       module_path, [&expected_file_size](const std::filesystem::path& cache_file_path) {
-        return VerifySymbolFile(cache_file_path, expected_file_size);
+        return VerifyFileSize(cache_file_path, expected_file_size);
       });
 }
 
+ErrorMessageOr<std::filesystem::path> SymbolHelper::FindObjectInCache(
+    const std::filesystem::path& module_path, const std::string& build_id,
+    uint64_t expected_file_size) const {
+  return FindSymbolsInCacheImpl(
+      module_path,
+      [&build_id,
+       expected_file_size](const std::filesystem::path& cache_file_path) -> ErrorMessageOr<void> {
+        OUTCOME_TRY(VerifyFileSize(cache_file_path, expected_file_size));
+        return VerifyObjectFile(cache_file_path, build_id);
```

Comment:
Since we want an exact match, checking the build id is extra safety. If we didn't, this change would simplify a bit because `VerifyObjectFile` wouldn't be completely necessary. Let me know what you think.

Reply:
I think I would leave it in. Orbit always requires the build id currently and only Mizar uses the file size. Only relying on file size feels a bit risky to me (even though it is probably fine in reality). I would rather improve the cache mechanism completely to support files without build id

Reply:
Sounds good!